app/src/main/java/org/oppia/android/app/translation/AppLanguageWatcherMixin.kt

```diff
@@ -61,19 +64,24 @@ class AppLanguageWatcherMixin @Inject constructor(
       appLanguageLocaleHandler.initializeLocale(defaultDisplayLocale)
     }
 
-    // TODO(#52): Hook this up properly to profiles, and handle the non-profile activity cases.
-    val profileId = ProfileId.getDefaultInstance()
-    val appLanguageLocaleDataProvider = translationController.getAppLanguageLocale(profileId)
-    val liveData = appLanguageLocaleDataProvider.toLiveData()
-    liveData.observe(
+    val currentUserProfileId = profileManagementController.getCurrentProfileId()
+
+    val activityLanguageLocaleDataProvider = when {
+      shouldUseSystemLanguage -> translationController.getSystemLanguageLocale()
+      currentUserProfileId == null -> translationController.getSystemLanguageLocale()
+      else -> translationController.getAppLanguageLocale(currentUserProfileId)
+    }
+
+    val liveData = activityLanguageLocaleDataProvider?.toLiveData()
+    liveData?.observe(
```

Comment:
```suggestion
    activityLanguageLocaleDataProvider.toLiveData().observe(
```

Can't be null anymore, I believe, so this can be simplified.

Reply:
UPdated.